metaphor/looker/lookml_parser.py

```diff
@@ -164,6 +97,175 @@ def from_dict(
+        # db.schema.table
+        full_name = source_name
+    else:
+        raise ValueError(f"Invalid source name {source_name}")
+
+    # Normalize dataset name by lower casing & dropping the quotation marks
+    full_name = full_name.replace('"', "").lower()
+
+    return EntityId(
+        EntityType.DATASET,
+        DatasetLogicalID(
+            name=full_name, platform=connection.platform, account=connection.account
+        ),
+    )
+
+
+def to_virtual_view_id(name: str, virtualViewType: VirtualViewType) -> EntityId:
+    """
+    converts a virtual view name and type into an Virtual View entity ID
+    """
+    return EntityId(
+        EntityType.VIRTUAL_VIEW,
+        VirtualViewLogicalID(name=name, type=virtualViewType),
+    )
+
+
+def _get_upstream_datasets(
+    view_name, raw_views: Dict[str, Dict], connection: Connection
+) -> Set[EntityId]:
+    # The source for a view can be specified via "sql_table_name" or "derived_table".
+    # When not specified, it's default to the same name as the view itself.
+    # https://docs.looker.com/reference/view-params/sql_table_name-for-view
+    raw_view = raw_views.get(view_name)
+    if raw_view is None:
+        raise ValueError(f"Refer to non-existing view {view_name}")
+
+    if "derived_table" in raw_view:
+        sql = raw_view["derived_table"].get("sql")
+        if sql is not None:
+            return _extract_upstream_datasets_from_sql(sql, raw_views, connection)
+
+        # TODO(1329): Add support for native derived tables
+        return set()
+
+    source_name = raw_view.get("sql_table_name", view_name)
+    return {_to_dataset_id(source_name, connection)}
+
+
+def _extract_upstream_datasets_from_sql(
+    sql: str, raw_views: Dict[str, Dict], connection: Connection
+) -> Set[EntityId]:
+    upstream: Set[EntityId] = set()
+    try:
+        tables = sql_metadata.Parser(sql).tables
+    except Exception as e:
+        logger.warn(f"Failed to parse SQL:\n{sql}\n\nError:{e}")
+        return upstream
+
+    for table in tables:
+        if table.endswith(".SQL_TABLE_NAME"):
+            # Selecting from another derived table
+            # https://docs.looker.com/data-modeling/learning-lookml/sql-and-referring-to-lookml
+            view_name = table.split(".")[0]
+            upstream.update(_get_upstream_datasets(view_name, raw_views, connection))
+        else:
+            upstream.add(_to_dataset_id(table, connection))
+
+    return upstream
+
+
+def _build_looker_view(
+    raw_view: Dict, raw_views: Dict[str, Dict], connection: Connection
+) -> VirtualView:
+    name = raw_view["name"]
+    view = LookerView(
+        label=raw_view.get("label", None),
+        source_datasets=[
+            str(ds) for ds in _get_upstream_datasets(name, raw_views, connection)
+        ],
+    )
+
+    if "extends" in raw_view:
+        view.extends = [
+            str(to_virtual_view_id(view, VirtualViewType.LOOKER_VIEW))
+            for view in raw_view["extends"]
```

Comment:
A more pythonic way is to drop the `if` and change this line to

```suggestion
            for view in raw_view.get("extends", [])
```

Reply:
It's not entirely equivalent as it'll set the `view.extends` field even if `raw_view` has no `extends`.

Reply:
This does reduce one line 👍 . But personally still prefer not to the set the field if there is no entry, more align with LookerML format.

Reply:
SGTM